examples/example_build_planet.py

```diff
@@ -81,86 +51,126 @@
-    #together with a helper to switch between them at 25 GPa. So we define a class of "mantle":
-
-    class mantle_rock(helpers.HelperLowHighPressureRockTransition):
-        def __init__(self):
-            helpers.HelperLowHighPressureRockTransition.__init__(self,25.e9,UM,LM)
-
-    #Isothermal mantles are boring, here we define a mantle that has a linear temperature profile, all we need
-    #
-    mantle = planet.Planet.LayerLinearTemperature("mantle", mantle_rock(), 6371.e3, 3000.0, 300.0, 200)
-
-    #Now we calculate the planet. Go BurnMan Go!
-    Plan = planet.Planet([inner_core, outer_core, mantle], verbose=True)
-
-    #Now we output the mass of the planet and moment of inertia
+    # FIRST: we must define the composition of the planet as individual layers.
+    # A layer is defined by 4 parameters: Name, min_depth, max_depth,and number of slices within the layer.
+    # Separately the composition and the temperature_mode need to set.
+    radius_planet = 6371.e3
+    # inner_core
+    inner_core = burnman.Layer("inner core", radius_planet=radius_planet,
+        max_depth=6371.e3, min_depth=5151.e3, n_slices=10)
+    inner_core.set_material(burnman.minerals.other.Fe_Dewaele())
+    
+    # The minerals that make up our core do not currently implement the thermal equation of state, so we will define the temperature with the Brown & Shankland geotherm.
+    inner_core.set_temperature_mode( 'user_defined',
+        burnman.geotherm.brown_shankland(inner_core.depths))
+
+    # outer_core
+    outer_core = burnman.Layer( "outer core", radius_planet=radius_planet,
+        max_depth=5151.e3, min_depth=2891.e3, n_slices=10)
+    outer_core.set_material(burnman.minerals.other.Liquid_Fe_Anderson())
+    # The minerals that make up our core do not currently implement the thermal equation of state, so we will define the temperature with the Brown & Shankland geotherm.
+    outer_core.set_temperature_mode('user_defined',
+        burnman.geotherm.brown_shankland(outer_core.depths))
+
+    # Next the Mantle.
+    lower_mantle = burnman.Layer( "lower_mantle", radius_planet=radius_planet,
+        max_depth=2891.e3, min_depth=660.e3, n_slices=10)
+    lower_mantle.set_material(burnman.minerals.SLB_2011.mg_bridgmanite())
+    lower_mantle.set_temperature_mode('adiabat')
+    upper_mantle = burnman.Layer( "upper_mantle", radius_planet=radius_planet,
+        max_depth=660e3, min_depth=0., n_slices=10)
+    upper_mantle.set_material(burnman.minerals.SLB_2011.forsterite())
+    upper_mantle.set_temperature_mode('adiabat', temperature_top =1200.)
+
+
+
+    # Now we calculate the planet.
+    Plan = burnman.Planet('earth_like',
+                          [inner_core, outer_core, lower_mantle, upper_mantle], verbose=True)
+    # Here we compute its state. Go BurnMan Go!
+    # (If we were to change composition of one of the layers, we would have to
+    # recompute the state)
+    Plan.set_state()
```

Comment:
I really like that this is a set_state now. Flows well with the rest of the functionality of BurnMan